lib/runtime/adapters.js

```diff
@@ -17,40 +18,69 @@ function register(type, module) {
+        return adapters[type];
+    }
+
+    if (adapter_config[type]) {
+        var adapter = load_adapter(type);
+        return adapter;
     }
 
-    return adapters[type];
+    // XXX this should include location info
+    throw errors.compileError('RT-INVALID-ADAPTER', {type: type});
 }
 
-// Load and initialize the adapter modules specified in the given config.
-function load(config) {
-    logger.debug('loading adapters');
-    _.each(config, function(options, module) {
-        try {
-            var module_path = options.path || module;
-            if (module_path[0] === '.') {
-                module_path = path.resolve(process.cwd(), module_path);
-            }
+// Load the adapter of the given type.
+function load_adapter(type) {
+    var options = adapter_config[type];
+    try {
+        // If the path isn't specified in the config, assume there is a global
+        // installation of the module named `juttle-xyz-adapter`.
+        var module_path = options.path || 'juttle-' + type + '-adapter';
+
+        // Any relative module paths should be resolved in the context of the
+        // process cwd, not relative to this module
+        if (module_path[0] === '.') {
+            module_path = path.resolve(process.cwd(), module_path);
+        }
 
-            var start = new Date();
-            var init = require(module_path);
+        var start = new Date();
+        var init = require(module_path);
 
-            var loaded = new Date();
-            var adapter = init(options);
+        var loaded = new Date();
+        var adapter = init(options);
 
-            var initialized = new Date();
+        if (adapter.name !== type) {
+            throw new Error('adapter name ', adapter.name, ' != type ', type);
+        }
+        var initialized = new Date();
+
+        logger.debug(adapter.name, 'adapter loaded in', (loaded - start), 'ms,',
+                     'initialized in', (initialized - loaded), 'ms');
+
+        register(adapter.name, adapter);
 
-            logger.debug(adapter.name, 'adapter loaded in', (loaded - start), 'ms,',
-                         'initialized in', (initialized - loaded), 'ms');
+        return adapter;
+    } catch (err) {
+        logger.error('error loading adapter ' + module + ': ' + err.message);
+        throw err;
+    }
+}
 
-            register(adapter.name, adapter);
-        } catch (err) {
-            logger.error('error loading adapter ' + module + ': ' + err.message);
-            throw err;
+// Add configuration for the specified adapters but don't initialize them until
+// they are actually used.
+function load(config) {
+    logger.debug('configuring adapters', _.keys(config).join(','));
+    _.each(config, function(options, type) {
+        // To be backwards-compatible with the old configuration format, if the
+        // adapter name is of the form `juttle-xyz-adapter`, register it as type
+        // `xyz`. This should be deprecated at some point in the future.
```

Comment:
IMHO we should just the switch now, while we don't have many external users and make the corresponding change in all our adapters and their docs now, otherwise, this comment will never go away :)